doozerlib/cli/release_gen_payload.py

```diff
@@ -502,155 +506,131 @@ def _inconsistency_annotation(self, inconsistencies):
-            envs = ar["extra"]["docker"]["config"]["config"].get("Env", [])
-            source_repo_entry = list(filter(lambda env: env.startswith("SOURCE_GIT_URL="), envs))
-            source_commit_entry = list(filter(lambda env: env.startswith("SOURCE_GIT_COMMIT="), envs))
-            if not source_repo_entry or not source_commit_entry:
-                continue  # this image doesn't have required environment variables. is it a dist-git only image?
-            source_repo = source_repo_entry[0][source_repo_entry[0].find("=") + 1:]  # SOURCE_GIT_URL=https://example.com => https://example.com
-            source_commit = source_commit_entry[0][source_commit_entry[0].find("=") + 1:]  # SOURCE_GIT_COMMIT=abc => abc
-            nvrs = repo_commit_nvrs.setdefault(source_repo, {}).setdefault(source_commit, set())
-            nvrs.add(record.build["nvr"])
-            nvr_images[record.build["nvr"]] = record.image
-
-        # Finally, look at the dict and print an error if one repo has 2 or more commits
-        mismatched_siblings = set()
-        for repo, commit_nvrs in repo_commit_nvrs.items():
-            if len(commit_nvrs) >= 2:
-                red_print("The following NVRs are siblings but built from different commits:")
-                for commit, nvrs in commit_nvrs.items():
-                    for nvr in nvrs:
-                        image = nvr_images[nvr]
-                        mismatched_siblings.add(image.distgit_key)
-                        red_print(f"{nvr}\t{image.distgit_key}\t{repo}\t{commit}")
-        return mismatched_siblings
+            if arch not in image_meta.get_arches():
+                # If this image is not meant for this architecture
+                members[tag_name] = None  # We still need a placeholder in the tag mapping
+                continue
 
+            if tag_name in members and not explicit:
+                # If we have already found an entry, there is a precedence we honor for
+                # "-alt" images. Specifically, if a imagemeta declares its payload tag
+                # name explicitly, it will take precedence over any other entries
+                # https://issues.redhat.com/browse/ART-2823
+                # This was tag not explicitly declared, so ignore the duplicate image.
+                continue
 
-def default_is_base_name(runtime: Runtime):
-    version = runtime.get_minor_version()
-    if runtime.assembly is None or runtime.assembly == 'stream':
-        return f'{version}-art-latest'
-    else:
-        return f'{version}-art-assembly-{runtime.assembly}'
+            archive_inspector = build_inspector.get_image_archive_inspector(brew_arch)
 
+            if not archive_inspector:
+                # This is no build for this CPU architecture for this build. Don't worry yet,
+                # it may be carried by an -alt image or not at all for non-x86 arches.
+                members[tag_name] = None
```

Comment:
I think this indicates an error, because if there's no build for this CPU architecture for this build, it should have been excluded from `image_meta.get_arches()`.